### DISCUSSION TITLE
Nightscout on Google Cloud switching to FreeDNS

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2441] [Show and tell]
If you are using Nightscout on Google Cloud, you can now switch to using FreeDNS instead of noip.com.
https://navid200.github.io/xDrip/docs/Nightscout/FreeDNS.html

Using noip.com requires updating your hostname once a month or Nightscout will stop working.
FreeDNS does not have that limitation.

If you experience any issues with this or any other aspects, please report in these discussions.

**Comment by cheshire42** [comment]
Starting fresh transferring my Heroku build over to Google. The FreeDNS option doesn't seem to be working. I keep hanging after I've configured everything and I'm not sure how to troubleshoot this.

**Reply by Navid200** [reply]
Leave transferring your database from Heroku to after this troubleshooting is done.
Do you have Nightscout working with noip.com in your virtual machine?

**Reply by cheshire42** [reply]
I do now. I had skipped the noip step, and I think that was the issue. I've currently got everything up and running on FreeDNS with no issues.

One additional question I have (that might be outside the scope of this) - I've got a Google domain that I'd like to host my nightscout site on, but when I point to it the upload fails. Any idea why?

**Comment by Bard09** [comment]
I'm using FreeDNS with the current instructions and it's working-- but unfortunately one of the devices I tend to use Nightscout on runs a work VPN with security that blocks sites on FreeDNS subdomains.

Is there an alternate pathway for managing DNS that could work?  If so, how would I do that?  For example I have a DynDNS account-- not sure if that's an option.

**Reply by Navid200** [reply]
@tzachi-dar Would you please have a look at this question?

**Comment by tzachi-dar** [comment]
@Bard09 are you sure that the problem is with freedns and not with the letsencrypt certifiacates?
If indeed the problem is with freedns, than you need to do the following:
1) Get a DynDns hostname.
2) make sure that the machine ip is registered in their dns system.
3) get a new certificate running sudo certbot --nginx -d "$hostname" --redirect --agree-tos --no-eff-email
replace hostname with the new name that you have received.

**Reply by timtheis** [reply]
I realize this thread is a bit old... but of interest to me. I see two options offered in this thread.
1) running sudo certbot --nginx -d "$hostname" --redirect --agree-tos --no-eff-email
2) edit the /etc/free-dns.sh file.
Option 2 is clear on the HOSTNAME... but the DIRECTURL seems to presume a certificate. I have the URL for DYNDNS update where I pass it the IP address. What is the nomenclature in the free-dns.sh file to specify the Nightscout host IP Address as a parameter?

**Reply by Navid200** [reply]
@timtheis You are more than welcome to use our product.
I call it a product because it is a product of our hard work.  It is free.  But, it is still a product.

But, it was never meant to satisfy everyone.
If our product is not good enough for you, I am more than happy to hear why and if possible, we will make changes.

But, if what you are after is extreme customization, then, very likely, our product is not for you.  And you don't have to use it.
But, these forums are meant to help the user community and us.

If you are after a community where you can discuss customizations to the limit, these forums are not it.
Using our limited resources for what they are not meant to address is not very fair.

On the other hand, if I have misunderstood you and you indeed need help with using our product, please forgive me and please explain exactly what we can help you with.

**Comment by Bard09** [comment]
@tzachi-dar I'll give that a try!  It's been a while since I messed with my Dyn account.  

Does Google Cloud change IPs over time, so you'd need to resynchronize the new IP with the non-FreeDNS DNS host, or is it "fixed" once created?

Re: why I'm dealing with this, my corporate laptop uses a tool called Mimecast-- I actually don't know what it's objecting to-- the domains or the certs-- but it definitely throws a browser error when you visit a FreeDNS subdomain URL (I tried multiple).

**Reply by tzachi-dar** [reply]
A general comment: Seems that some of the domains used by freedns are classified as porn by some tools.
So, some would tools block them. One can probably also move to one of the less popular ones on freedns.

Google cloud might change an IP after a reboot. The way to deal with it is changed by the different providers.
If your provider has a URL that needs to be called to make the change, you can put it in the file: 
/etc/free-dns.sh

**Reply by Bard09** [reply]
@tzachi-dar Just FYI for @tzachi-dar and others-- I tried several other entries from the Shared Domain Registry (https://freedns.afraid.org/domain/registry/) and eventually found one that worked with my corporate security policy.  If you have the same problem I did, keep trying other subdomain options!

**Comment by emp-00** [comment]
I have switched from noip.com to FreeDNS as per @Navid200 's instructions one week ago. Ever since both services are running -> obviously nightscout is still connected to noip.com and still updates those dns entries as well. Q: How can I completely disconnect nightscout from noip.com? Of course I can disable my noip.com account but then I guess nightscout will still try to connect to this service. I think we need to be able to disable noip.com completely to avoid unnecessary connections/errors. Maybe I'm wrong? Was this done via the FreeDNS script and maybe something went wrong in my case?

**Reply by Navid200** [reply]
@tzachi-dar

**Reply by tzachi-dar** [reply]
nightscout is not realy connected to no-ip. This works the other way around, that is no-ip points to your nightscout site. You can simply close your no-ip account. Alternatively, no-ip just sends you a mail every month, just ignore that mail and your account will be closed automatically.

**Reply by Navid200** [reply]
You don't have to close the account, do you?  I mean you can just delete the hostname.

**Comment by bjornoleh** [comment]
FYI:
Someone had troubles with freedns denying to give certificates, too many already issued to .strangled.net:

![D261FE85-CC8D-4290-89C4-9BB70138C099](https://user-images.githubusercontent.com/63544115/205356700-6222dd51-c15a-4a58-b743-a9ff59098f76.jpeg)

I guess many will bump into this now. 

https://discord.com/channels/629952586895851530/629952669967974410/1048288940098076813

**Comment by kachytronico** [comment]
Yes, I had problems even though I changed servers. 

![Captura de pantalla 2022-12-03 a las 1 08 28](https://user-images.githubusercontent.com/1207027/205411661-3fd5206c-2635-4f87-9b49-a4290d845cf8.png)

**Reply by bjornoleh** [reply]
Did it resolve in the end? Anything specific you needed to do?

**Comment by tzachi-dar** [comment]
Either select a new name from freedns or wait 24 hours and try again.

**Comment by RyRyCeratops** [comment]
I had the same problem with the certificates. It actually gives you a date and time to try again.  I waited and then it worked for me.  I did not have to wait 24 hours - only about 5 hours. It tells you in the red text when you can try again - date and time.

**Reply by Navid200** [reply]
One thing to consider is that if you pick one of the 7 domains shown at the top when you go to subdomains, it is possible to run into trouble.  The reason is that most people think that those 7 are their only options or just don't care to spend a few seconds to pick a non-popular domain.  That explains the error: Too many certificates ....

The more users you have on a domain, the more likely it is that one of them is going to use it for a purpose causing the domain to be tagged by scanners as high-risk and blocked by routers.  So, this is another reason to avoid popular domain names.

I have explained this in detail here: https://navid200.github.io/xDrip/docs/Nightscout/FD_Domains.html

**Comment by RyRyCeratops** [comment]
I do not understand why I was copied here. My Nightscout also stopped working but I pay a monthly fee to have it work and I very happy with it. I do not know what any of the below means and I have not sent any messages. Can you please help explain. That @ below is not me. I do not know who that is. Please help if you can. Thank you! AmySent from my iPhoneOn Dec 16, 2024, at 3:40 PM, Navid ***@***.***> wrote:﻿
@timtheis You are more than welcome to use our product.
I call it a product because it is a product of our hard work.  It is free.  But, it is still a product.
But, it was never meant to satisfy everyone.
If our product is not good enough for you, I am more than happy to hear why and if possible, we will make changes.
But, if what you are after is extreme customization, then, very likely, our product is not for you.  And you don't have to use it.
But, these forums are meant to help the user community and us.
If you are after a community where you can discuss customizations to the limit, these forums are not it.
Using our limited resources for what they are not meant to address is not very fair.
On the other hand, if I have misunderstood you and you indeed need help with using our product, please forgive me and please explain exactly what we can help you with.

**Reply by Navid200** [reply]
@RyRyCeratops When you post in a discussion on GitHub, you automatically subscribe to that discussion.  Then, as soon as anyone makes a post in that discussion, you get a notification.
You can see in the note you got that I tagged (@ symbol) the individual who posted in this very old thread to respond to him.
You had posted in this old thread in 2023 and that's how you subscribed to it.

You can unsubscribe.
It is almost at the top in the right pane.
